src/entities/fractions/fraction.rs

```diff
@@ -253,20 +257,23 @@ mod tests {
     #[test]
     fn test_invert() {
         let fraction = Fraction::new(5, 10).invert();
-        assert_eq!(fraction.numerator, BigInt::from(10));
-        assert_eq!(fraction.denominator, BigInt::from(5));
+        assert_eq!(fraction.clone().numerator, BigInt::from(10));
+        assert_eq!(fraction.clone().denominator, BigInt::from(5));
     }
 
     #[test]
     fn test_add() {
-        assert_eq!(
-            Fraction::new(1, 10) + Fraction::new(4, 12),
-            Fraction::new(52, 120)
-        );
-        assert_eq!(
-            Fraction::new(1, 5) + Fraction::new(2, 5),
-            Fraction::new(3, 5)
-        );
+        let fraction1 = Fraction::new(1, 10);
+        let fraction2 = Fraction::new(4, 12);
+        let expected = Fraction::new(52, 120);
+        let add = fraction1 + fraction2;
+        assert_eq!(add, expected);
+
+        let fraction3 = Fraction::new(1, 5);
+        let fraction4 = Fraction::new(2, 5);
+        let expected2: FractionLike<()> = Fraction::new(3, 5);
+        let add2 = fraction3 + fraction4;
+        assert_eq!(add2, expected2);
```

Comment:
What's wrong with the original test?

Reply:
nothing, the changes I made were to handle result type before, now that's it's no longer result type, will abastract the tests when during the next pr